http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -184,16 +184,19 @@ class DataprocInterpreter[F[_]: Parallel](
           .compile
           .drain
 
-        // build cluster configuration
-        initScriptResources = List(config.clusterResourcesConfig.initScript)
-        initScripts = initScriptResources.map(resource => GcsPath(initBucketName, GcsObjectName(resource.asString)))
-
-        // If we need to support 2 version of dataproc custom image, we'll update this
-//        dataprocImage = config.dataprocConfig.customDataprocImage
+        imageUrls = params.runtimeImages.map(_.imageUrl)
 
+        // build cluster configuration
         // We need to maintain the old version of the dataproc image to uncouple the terra from the aou release
-        imageUrls = params.runtimeImages.map(_.imageUrl)
-        dataprocImage = config.dataprocConfig.customDataprocImage
+        // AN-503: Delete once AOU has switched to using Dataproc 2.2.X in prod
+        (dataprocImage, initScriptResources) =
+          if (imageUrls.contains("us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.13"))
```

Comment:
oof they're far behind. Should this also be checking if the imageUrls contain any `terra-jupyter-aou` image before `2.2.16` or is the plan to jump from 13 --> 16?

Reply:
We only cache 2.2.13 and 2.2.16 now, so any other version will time out. I borrowed this code from an old PR of mine when we did something similar going from dataproc 2.0.x to 2.1.x so I am comfortable with having this for a month or so if you are okay with it

Reply:
yeah that makes sens, thanks for adding all the ticketed comments for when that happens!